pkg/datapath/loader/base.go

```diff
@@ -499,3 +500,50 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 
 	return nil
 }
+
+// OnUpdateCiliumNode is only called if running with IPAM ENI and IPsec. In
+// that case, we need to rerun reinitializeIPSec whenever new ENIs are added to
+// the local node, to attach our BPF program for IPsec decryption to the new
+// interfaces.
+func (l *Loader) OnUpdateCiliumNode(oldNode, newNode *ciliumv2.CiliumNode, _ *lock.StoppableWaitGroup) error {
```

Comment:
I'd like to solve this in a cleaner way (and without k8s dependencies in pkg/datapath) by having loader observe the devices table (coming in https://github.com/cilium/cilium/pull/24677). I'm fine with this workaround until the proper ground work is done. Is there any pitfalls to just watching for the local network devices and addresses versus the CiliumNode object?

Reply:
> Is there any pitfalls to just watching for the local network devices and addresses versus the CiliumNode object?

I think that would work. Given Sebastian's comment below, it would probably even be easier once we have the new controller in place.